aea/agent.py

```diff
@@ -310,3 +309,20 @@ def get_periodic_tasks(
         :return: dict of callable with period specified
         """
         return {self.act: (self._timeout, None)}
+
+    def _handle_envelope(self, envelope: Envelope) -> None:
```

Comment:
why not rather make it an `@abstractmethod`?

Reply:
done